roles/libvirt_manager/tasks/create_networks.yml

```diff
@@ -81,6 +82,112 @@
   loop_control:
     label: "{{ item.name }}"
 
+## Ensure we get dnsmasq DHCP service for all created networks
+# Refreshing network facts will ensure we get the new interfaces.
+# They (usually?) have the same name as the network itself.
+- name: Refresh networking facts on host
+  ansible.builtin.setup:
+    gather_subset:
+      - '!all,!min'
+      - network
+
+- name: Build needed network related content
+  when:
+    - ansible_facts[_name] is defined
+  vars:
+    _name: "{{ item | replace('-', '_') }}"
+    _no_prefix_name: "{{ _name | regex_replace('cifmw[-_]','') }}"
+    _fixed_nets: >-
+      {{
+        cifmw_libvirt_manager_fixed_networks_defaults +
+        cifmw_libvirt_manager_fixed_networks
+      }}
+    _dns_options: |-
+      {% if ansible_facts[_name].ipv4 is defined -%}
+      - "option:dns-server,{{ ansible_facts[_name].ipv4.address }}"
```

Comment:
This assumes the DNS server for a network is the address of the network itself. I am not convinced this is always what we want.